lab_5_scrapper/scrapper.py

```diff
@@ -90,25 +204,41 @@ def __init__(self, config: Config) -> None:
         """
         Initializes an instance of the Crawler class
         """
-        pass
+        self.urls = []
+        self.config = config
+        self.seed_urls = config.get_seed_urls()
+        self.number_of_articles = config.get_num_articles()
 
     def _extract_url(self, article_bs: BeautifulSoup) -> str:
         """
         Finds and retrieves URL from HTML
         """
-        pass
+        all_links_bs = article_bs.find_all('a')
+        for link in all_links_bs:
+            href = link.get('href')
+            total_digits = len(re.findall('[0-9]', href))
```

Comment:
strange condition

Reply:
some of the false urls start with '/news/' and can have the same number of slashes as the correct urls, so i check for a number of digits as correct urls always contain a date there (8 digits)